net-dns/yaddns/yaddns-1.1.recipe

```diff
@@ -0,0 +1,78 @@
+SUMMARY="A small dynamic DNS client"
+DESCRIPTION="Yaddns is a small dynamic dns client with multiple servers \
+support and a high flexibility."
+HOMEPAGE="https://yaddns.github.io/"
+COPYRIGHT="2015 Anthony Viallard and Raphael Huck"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/yaddns/yaddns/archive/yaddns-$portVersion.tar.gz"
+CHECKSUM_SHA256="23b527b7f71e1449746f1e39cbc3a074d58bd261a33399fb4e2d90eb63149059"
+SOURCE_DIR="yaddns-yaddns-$portVersion"
+PATCHES="yaddns-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
+
+PROVIDES="
+	yaddns$secondaryArchSuffix = $portVersion
+	cmd:yaddns$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:autom4te
+	cmd:automake
+	cmd:autoreconf
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	"
+
+USER_SETTINGS_FILES="
+	settings/yaddns directory
```

Comment:
```diff
 	cmd:makeinfo
 	"
 
-USER_SETTINGS_FILES="
-	settings/yaddns directory
-	"
-
 PATCH()
```
(We can drop this because we now put the file `yaddns.conf` in `/system/settings/` instead of `/system/settings/yaddns/`).

BTW, thanks for the update!
It looks like everything is OK. I'm checking this right now.

Reply:
Do we need to add `yaddns.conf` itself?

Reply:
Yes, you would need to @sfanxiang

Reply:
Hmm, to avoid any confusion, let me rephrase what @alaviss wrote:

We don't need to ship any `yaddns.conf` (other than `yaddns.conf.example`).
It's the user who will need to create that file, in `/system/settings/`.
But this is OK like this :)

Reply:
@fbrosson I see, thanks. My mind was still stuck with the idea that `yaddns.conf` should be included after we already determined otherwise.

BTW, if a conf file is "good" enough, can we add it?

Reply:
@fbrosson we can't drop `USER_SETTINGS_FILE`. It's supposed to list whatever system configuration file that the port uses, which must not be included within the package (no ref for now as PackageManagement section on Trac magically disappeared)

Reply:
Ref: https://web.archive.org/web/20160625111930/https://dev.haiku-os.org/wiki/PackageManagement/BuildingPackages

> user-settings-files:
> ...
> path is the relative path of the settings file or directory, starting with "settings/". It is __not__ included in the package.

Reply:
As a conclusion, is this correct:
- the actual file `yaddns.conf` shouldn't be shipped with yaddns, and
- the file name `yaddns.conf` should be added to `USER_SETTINGS_FILE`

?

---

But @alaviss I can't find where it says `path` __must not__ be included, and I think
> It is not included in the package.

wants to indicate that `path` is not automatically regarded as an included file, instead of forbidding `path` to be included. Otherwise, where do all those configuration files come from when Haiku is installed?

Reply:
I you are perfectly right, @alaviss. Thanks for spotting this. I though it was not needed, but it does perfectly make sense to have a `USER_SETTINGS_FILES`.
I've just added:
```
USER_SETTINGS_FILES="
	settings/yaddns.conf
	"
```
and haikuporter is happy with it :)

Reply:
> BTW, if a conf file is "good" enough, can we add it?

@sfanxiang: Yes, and it is even a good thing. But in such cases we install both the default config file and its template. And both need to be identical.

Guess what? I was going to add a pointer to a recipe that was recently added and which has a default config and its template, and I just noticed that you are the author of that recipe: [app-crypt/p11-kit/](https://github.com/haikuports/haikuports/tree/master/app-crypt/p11-kit)

For yaddns, however, since the example config is not usable as is, we would need to modify it.
But the only acceptable default config would be a config with no active content. So if we change the example config to make it acceptable as a default config, it looses its interest.

Well, we could have an inactive default config with its identical template, and an example config. But I think this would not be a good idea.
I think we can assume that anyone willing to use a dyndns service can start from the example config and adjust it, with the manual page, to have a correct config.